libraries/Adapters/Microsoft.Bot.Builder.Adapters.Webex/WebexAdapter.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Builder.BotKit.Core;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Thrzn41.WebexTeams.Version1;
+
+namespace Microsoft.Bot.Builder.Adapters.Webex
+{
+    public class WebexAdapter : BotAdapter
+    {
+        /// <summary>
+        /// Name used by Botkit plugin loader.
+        /// </summary>
+        public const string Name = "Webex Adapter";
+
+        private readonly IWebexAdapterOptions config;
+
+        private readonly TeamsAPIClient api;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebexAdapter"/> class.
+        /// Creates a Webex adapter. See [WebexAdapterOptions] for a full definition of the allowed parameters.
+        /// </summary>
+        /// <param name="options">An object containing API credentials, a webhook verification token and other options.</param>
+        public WebexAdapter(IWebexAdapterOptions options)
+            : base()
+        {
+            this.config = {};
+
+            if (this.config.AccessToken != null)
+            {
+                this.api = new TeamsAPIClient(this.config.AccessToken);
+
+                if (this.api == null)
+                {
+                    throw new Exception("Could not create the Webex Teams API client");
+                }
+            }
+            else
+            {
+                // error: access_token required to create controller
+            }
+
+            if (this.config.PublicAdress != null)
+            {
+                var endpoint =  this.config.PublicAdress;
+            }
+            else
+            {
+                // error: public_address parameter required to receive webhooks
+            }
+
+            if (this.config.Secret != null)
```

Comment:
I think this condition should be evaluating for null

Reply:
why would that be? 
https://github.com/southworkscom/botkit/blob/master/packages/botbuilder-adapter-webex/src/webex_adapter.ts#L360

Reply:
Is this line:
https://github.com/southworkscom/botkit/blob/master/packages/botbuilder-adapter-webex/src/webex_adapter.ts#L138
You should throw an exception if the Secret is null

Reply:
The secret is an optional parameter. Check the Node Js version